frontend/packages/shared/src/components/InputActionWrapper/InputActionWrapper.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from 'react';
+import { CheckmarkIcon, TrashIcon, PencilWritingIcon } from '@altinn/icons';
+import { useText } from '../../../../ux-editor/src/hooks/index';
+import { Button } from '@digdir/design-system-react';
+import classes from './InputActionWrapper.module.css';
+
+type AvailableAction = 'edit' | 'save' | 'delete';
+export type ActionGroup = 'editMode' | 'hoverMode' | 'standBy';
+
+const actionGroupMap: Record<ActionGroup, AvailableAction[]> = {
+  editMode: ['save', 'delete'],
+  hoverMode: ['edit', 'delete'],
+  standBy: [],
+};
+
+const actionToIconMap: Record<AvailableAction, React.ReactNode> = {
+  edit: <PencilWritingIcon />,
+  save: <CheckmarkIcon />,
+  delete: <TrashIcon />,
+};
+
+export type InputActionWrapperProps = {
+  children: React.ReactElement;
+  mode?: ActionGroup;
+  onEditClick: () => void;
+  onDeleteClick: () => void;
+  onSaveClick: () => void;
+};
+
+export const InputActionWrapper = ({
+  mode,
+  children,
+  onEditClick,
+  onDeleteClick,
+  onSaveClick,
+  ...rest
+}: InputActionWrapperProps): JSX.Element => {
+  const t = useText();
+  const [actions, setActions] = useState<AvailableAction[]>(actionGroupMap[mode || 'standBy']);
+
+  const handleFocus = (): void => {
+    setActions(actionGroupMap['editMode']);
+  };
+
+  const handleBlur = (): void => {
+    setActions(actionGroupMap['standBy']);
+  };
+
+  const handleMouseLeave = (): void => {
+    const isEditMode = actions.includes('save');
+    if (isEditMode) return;
+    handleBlur();
+  };
+
+  const handleActionClick = (action: AvailableAction): void => {
+    switch (action) {
+      case 'delete':
+        onDeleteClick();
+        break;
+      case 'save':
+        onSaveClick();
+        handleBlur();
+        break;
+      case 'edit':
+        onEditClick();
+        setActions(actionGroupMap['editMode']);
+      default:
+        break;
+    }
+  };
+
+  const handleHover = (): void => {
+    const isInStandByMode = mode === 'standBy';
+    if (isInStandByMode) {
+      setActions(actionGroupMap['hoverMode']);
+    }
+  };
+
+  // Handled Variant, size, and color in functions if we need to change them based on status  in the future.
+  const actionToAriaLabelMap: Record<AvailableAction, string> = {
+    edit: t('general.edit'),
+    delete: t('general.delete'),
+    save: t('general.save'),
+  };
+
+  const handleVariant = (action: AvailableAction): 'filled' | 'quiet' | 'outline' => {
+    return 'quiet';
+  };
+
+  const handleColor = (action: AvailableAction): 'danger' | 'primary' | 'secondary' | 'success' => {
```

Comment:
I think you should change this to a map instead of a method. What do you think about that? 😊 
```suggestion
  const actionToColorMap: Record<AvailableAction, ButtonProps['color']> = {
     'edit': 'primary',
  }
```